components/SingleClassView/Module/Module.tsx

```diff
@@ -0,0 +1,400 @@
+import React, { useContext, useEffect, useState } from "react";
```

Comment:
github stopped rendering this dif, so i think this file is too big. Lets split these into different components, I can do it if u would like (jetbrains makes it easy af)

Reply:
please do if you can <3